Apps/WebClient/src/ClientApp/src/components/services/OrganDonorDetailsCard.vue

```diff
@@ -0,0 +1,183 @@
+﻿<script lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCircleInfo, faDownload } from "@fortawesome/free-solid-svg-icons";
+import saveAs from "file-saver";
+import Vue from "vue";
+import { Component, Prop, Ref } from "vue-property-decorator";
+import { Action, Getter } from "vuex-class";
+
+import MessageModalComponent from "@/components/modal/MessageModalComponent.vue";
+import PatientData, {
+    HealthOptionTypes,
+    OrganDonorRegistrationData,
+    PatientDataFile,
+    PatientHealthOptions,
+} from "@/models/patientData";
+import container from "@/plugins/container";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger } from "@/services/interfaces";
+import SnowPlow from "@/utility/snowPlow";
+
+library.add(faCircleInfo, faDownload);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const options: any = {
+    components: { MessageModalComponent },
+};
+
+@Component(options)
+export default class OrganDonorDetailsCard extends Vue {
+    @Prop({ required: true })
+    hdid!: string;
+
+    @Getter("patientData", { namespace: "patientData" })
+    getPatientData!: (hdid: string) => PatientData;
+
+    @Action("retrievePatientDataFile", { namespace: "patientData" })
+    getPatientDataFile!: (params: {
+        fileId: string;
+        hdid: string;
+    }) => Promise<PatientDataFile>;
+
+    @Ref("sensitiveDocumentModal")
+    readonly sensitiveDocumentModal!: MessageModalComponent;
+
+    logger!: ILogger;
+
+    get registrationData(): OrganDonorRegistrationData | undefined {
+        return this.patientData.items.find(
+            (ho: PatientHealthOptions) =>
+                ho.type === HealthOptionTypes.OrganDonorRegistrationData
+        ) as OrganDonorRegistrationData;
+    }
+
+    get patientData(): PatientData {
+        return this.getPatientData(this.hdid);
+    }
+
+    private created(): void {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    }
+
+    private showConfirmationModal(): void {
+        this.sensitiveDocumentModal.showModal();
+    }
+
+    private getDecisionFile(): void {
+        if (this.registrationData && this.registrationData.registrationFileId) {
+            SnowPlow.trackEvent({
+                action: "download_report",
+                text: "Organ Donor",
```

Comment:
Recommend using "organ_donor" but should confirm with product owner.

Reply:
This was in the story, I'll check what the other text values for similar actions.

Reply:
Similar text values are not snake case.

Reply:
I guess the "download_report" ones are.  It's very inconsistent.